server.py

```diff
@@ -12,21 +12,56 @@ def server():
     try:
         while True:
             conn, addr = server_socket.accept()
-            buffsize = 32
+            buffsize = 16
             msg = ""
             complete = False
             while not complete:
                 line = conn.recv(buffsize)
-                print line
+                msg = "{}{}".format(msg, line)
                 if len(line) < buffsize:
                     complete = True
-                msg = "{}{}".format(msg, line)
-            conn.sendall("You sent: {}".format(msg))
-            print "You received: {}".format(msg)
+            response = parse_request(msg)
+            try:
+                conn.sendall(response)
+            except TypeError:
+                conn.sendall(response_error('400', 'BAD REQUEST'))
             conn.close()
     except KeyboardInterrupt:
+        print "\nServer successfully shut down"
         server_socket.close()
 
 
+def response_ok(msg):
+    result = "HTTP/1.1 200 OK\r\n"
+    con_type = "Content-Type: text/plain\r\n"
+    body = "Content length: {}".format(len(msg))
+    return "{}{}{}".format(result, con_type, body)
```

Comment:
You've jumped a bit ahead here and gotten yourself in a twist in the process.  The `response_ok` function at this point should simply return a statically generated 200 response with a simple body.  Something like this:

```
HTTP/1.1 200 OK
Content-Type: text/plain
Content Length: 22

Everything is Awesome!
```

You are trying to calculate things dynamically but we haven't gotten to that part yet, and as a result you are not yet building a proper http response.  Let's get a single static response right before trying to get fancy.